Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -167,7 +176,43 @@ public class Bip32Ed25519 {
         return (z, childChainCode)
     }
 
-    func deriveChildNodePrivate(extendedKey: Data, index: UInt32) -> Data {
+    func deriveChildNodePublic(extendedKey: Data, keyIndex: UInt32, g: Int = 9) throws -> Data {
+        guard keyIndex < 0x8000_0000 else {
+            throw DataValidationException(message: "Cannot derive public key with harden")
+        }
+
+        let pk = extendedKey.subdata(in: 0 ..< 32)
```

Comment:
We can introduce a constant called ED25519_POINT_SIZE = 32 and replace the values here, equivalent to the ED25519_SCALAR_SIZE.